src/core/mac/sub_mac.cpp

```diff
@@ -723,12 +809,154 @@ const char *SubMac::StateToString(State aState)
     case kStateEnergyScan:
         str = "EnergyScan";
         break;
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    case kStateCslTransmit:
+        str = "CslTransmit";
+        break;
+#endif
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+    case kStateCslSample:
+        str = "CslSample";
+        break;
+#endif
     }
 
     return str;
 }
 
 // LCOV_EXCL_STOP
 
+//---------------------------------------------------------------------------------------------------------------------
+// CSL Receiver methods
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+void SubMac::SetCslChannel(uint8_t aChannel)
+{
+    mCslChannel = aChannel;
+}
+
+void SubMac::SetCslPeriod(uint16_t aPeriod)
+{
+    mCslPeriod = aPeriod;
+}
+
+void SubMac::SetCslTimeout(uint32_t aTimeout)
+{
+    mCslTimeout = aTimeout;
+}
+
+void SubMac::FillCsl(Frame &aFrame)
+{
+    uint8_t *cur = aFrame.GetHeaderIe(Frame::kHeaderIeCsl);
+
+    if (cur != NULL)
+    {
+        CslIe *csl = reinterpret_cast<CslIe *>(cur + sizeof(HeaderIe));
+
+        csl->SetPeriod(mCslPeriod);
+        csl->SetPhase(GetCslPhase());
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+        otLogDebgMac("%10lu:%s seq=%u phase=%hu", static_cast<uint32_t>(otPlatTimeGet()), __func__,
```

Comment:
Suggest avoid using `__func__` since its behavior is tool-chain dependent. 
```suggestion
        otLogDebgMac("%10lu:SetCsl() seq=%u phase=%hu", static_cast<uint32_t>(otPlatTimeGet()),
```